From 08a0a4701143a1001c90501ce8d4ebb3c1c56900 Mon Sep 17 00:00:00 2001
From: Hans Christian Lonstad <hcl@datarespons.no>
Date: Fri, 26 Aug 2022 10:31:51 +0200
Subject: [PATCH 4/5] mxxf1: Only ctl-c should halt boot, add version info

Testing reveals that button events are induced from external cable
connected push buttons. This may unintentionally halt the boot
process.

Add patch level and comit SHA to version string in NVRAM
---
 arch/arm/boards/dr-mxxf1/env/init/greeting |  1 +
 arch/arm/boards/dr-mxxf1/vpd-eeprom.c      | 10 ++++++----
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boards/dr-mxxf1/env/init/greeting b/arch/arm/boards/dr-mxxf1/env/init/greeting
index 0000c30bc9..56c372be36 100644
--- a/arch/arm/boards/dr-mxxf1/env/init/greeting
+++ b/arch/arm/boards/dr-mxxf1/env/init/greeting
@@ -5,3 +5,4 @@ fbconsole0.active=oe
 echo "DR MXXF1 Barebox: $global.version"
 echo "Board REV is $global.board.revision"
 echo "Reset reason is $global.system.reset"
+global.autoboot_abort_key="ctrl-c"
diff --git a/arch/arm/boards/dr-mxxf1/vpd-eeprom.c b/arch/arm/boards/dr-mxxf1/vpd-eeprom.c
index eb54c4ca51..7fa25f273e 100644
--- a/arch/arm/boards/dr-mxxf1/vpd-eeprom.c
+++ b/arch/arm/boards/dr-mxxf1/vpd-eeprom.c
@@ -67,17 +67,19 @@ static int vpd_update_eeprom(void)
 	const char *val;
 	int status = 0;
 	int dirty = 0;
+	static char vstr[256];
 	int index = param_find(&ee, vkey);
+	sprintf(vstr, "barebox-%s", buildsystem_version_string);
 	if (index >= 0) {
 		val = param_get(&ee, vkey);
-		if (IS_ERR_OR_NULL(val) || strcmp(val, release_string) != 0) {
+		if (IS_ERR_OR_NULL(val) || strcmp(val, vstr) != 0) {
 			pr_debug("%s: Update existing key %s\n", __func__, vkey);
-			status = param_set(&ee, vkey, release_string);
+			status = param_set(&ee, vkey, vstr);
 			dirty = 1;
 		}
 	} else {
 		pr_debug("%s: No key %s present\n", __func__, vkey);
-		status = param_set(&ee, vkey, release_string);
+		status = param_set(&ee, vkey, vstr);
 		dirty = 1;
 	}
 
@@ -100,7 +102,7 @@ static int vpd_update_eeprom(void)
 	index = param_find(&ee, vkey);
 	if (index >= 0) {
 		val = param_get(&ee, vkey);
-		if (IS_ERR_OR_NULL(val) || strcmp(val, release_string) != 0) {
+		if (IS_ERR_OR_NULL(val) || strcmp(val, vstr) != 0) {
 			pr_err("EEPROM not updated (%d)\n", (int)PTR_ERR(val));
 			return (int)PTR_ERR(val);
 		}
-- 
2.34.1

